pkgs/os-specific/linux/sgx-sdk/default.nix

```diff
@@ -19,59 +18,63 @@
 , perl
 , python3
 , texinfo
-, which
+, validatePkgConfig
 , writeShellScript
+, writeText
 }:
+with lib;
 
 stdenv.mkDerivation rec {
   pname = "sgx-sdk";
-  version = "2.14";
+  version = "2.14.100.2";
 
   src = fetchFromGitHub {
     owner = "intel";
     repo = "linux-sgx";
-    rev = "0cea078f17a24fb807e706409972d77f7a958db9";
+    rev = "sgx_${concatStringsSep "." (take 2 (splitVersion version))}";
     sha256 = "1cr2mkk459s270ng0yddgcryi0zc3dfmg9rmdrdh9mhy2mc1kx0g";
     fetchSubmodules = true;
   };
 
-  patches = [
-    (fetchpatch {
-      name = "replace-bin-cp-with-cp.patch";
-      url = "https://github.com/intel/linux-sgx/commit/e0db5291d46d1c124980719d63829d65f89cf2c7.patch";
-      sha256 = "0xwlpm1r4rl4anfhjkr6fgz0gcyhr0ng46fv8iw9hfsh891yqb7z";
-    })
-    (fetchpatch {
-      name = "sgx_ippcp.h.patch";
-      url = "https://github.com/intel/linux-sgx/commit/e5929083f8161a8e7404afc0577936003fbb9d0b.patch";
-      sha256 = "12bgs9rxlq82hn5prl9qz2r4mwypink8hzdz4cki4k4cmkw961f5";
-    })
-  ];
+  postUnpack = ''
+    # Make sure this is the right version
+    grep -q '"${version}"' "$src/common/inc/internal/se_version.h" \
+      || (echo "Could not find expected version ${version} in linux-sgx source" >&2 && exit 1)
+  '';
+
   postPatch = ''
-    patchShebangs ./linux/installer/bin/build-installpkg.sh \
-      ./linux/installer/common/sdk/createTarball.sh \
-      ./linux/installer/common/sdk/install.sh
+    # https://github.com/intel/linux-sgx/pull/730
+    substituteInPlace buildenv.mk --replace '/bin/cp' 'cp'
+
+    # https://github.com/intel/linux-sgx/pull/752
+    ln -s "$src/external/epid-sdk/ext/ipp/include/sgx_ippcp.h" \
+          'external/ippcp_internal/inc/sgx_ippcp.h'
```

Comment:
Cool! I did not know that the missing header file was under this `external/epid-sdk`.

Just wish to note here that doing `diff` between this `external/epid-sdk/ .../sgx_ippcp.h` and the one that is packaged in "Prebuilt Optimized Libraries Binaries" at https://01.org/intel-softwareguard-extensions/downloads/intel-sgx-linux-2.14-release, does not match. After inspection, it can be seen that it's only the formatting and comments that differ. 

However, it appears that this `external/epid-sdk/ext/ipp/` is not kept up-to-date. It has not been updated since release 2.13.3, and in the case of 2.14, it may just happen that we're lucky that the header file (`sgx_ippcp.h`) did not change, but it does in 2.15. I do not know what is the significance of these changes, but just wish to point out that this approach may be somewhat fragile, as future updates to the derivation for newer sgx sdk versions, will require a different approach if the source repo (intel/linux-sgx) is not updated to match what is published/packaged by Intel.

It would be best to see with Intel, to make sure that they keep the header file in sync with what they publish for download. Opened and issue: https://github.com/intel/linux-sgx/issues/759.

In the meantime, it seems to me that sticking to the patch/commit/PR would be simpler, and easier to double-check if needed, but I don't mind if you prefer the newer proposed approach. :)

Reply:
Yes, I also inspected the diff of your patch and the version from `epid-sdk` and concluded that they match.

In the end, it doesn't really matter if we add the missing file (your approach) or if we copy/link it (my approach). Ultimately, both approaches are a patch and we have to keep it in sync with the corresponding versions. Therefore, it makes a lot of sense that you already opened upstream issues.

I'm not against using a patch. For the current version it just made sense to use the version we already have in the source derivation. I'm fine with going back to a patch but I'd prefer adding the patch file (i.e., check it in).

Maybe it's most idiomatic to use `fetchurl` to fetch the `sgx_ippcp.h` file from the original source were you created the patch from?

Reply:
> Maybe it's most idiomatic to use `fetchurl` to fetch the `sgx_ippcp.h` file from the original source were you created the patch from?

See https://github.com/NixOS/nixpkgs/pull/146058/commits/406fb9b14db226fed9b49d6e95b0373683b2a586.

Reply:
Oh thanks for going through the effort for this different approach. I think it makes it easier to quickly double check that the header is the expected one, but perhaps that is just a personal preference. Either way, I like the new approach, and agree with you that either way it is temporary as we would prefer upstream to take proper care of this. :+1: